shaft_element.py

```diff
@@ -1,478 +1,461 @@
-        geometry, materials = load_shaft_from_xltrc(file, sheet_name)
-        shaft = [
-            ShaftElement(
-                el.length,
-                el.id_Left,
-                el.od_Left,
-                materials[el.matnum],
-                n=el.elemnum - 1,
-            )
-            for i, el in geometry.iterrows()
-        ]
-
-        return shaft
-
-        #  TODO stiffness Matrix due to an axial load
-        #  TODO stiffness Matrix due to an axial torque
-        #  TODO add speed as an argument so that skew-symmetric stiffness matrix can be evaluated (default to None)
-        #  TODO skew-symmetric speed dependent contribution to element stiffness matrix from the internal damping
-        #  TODO add tappered element. Modify shaft element to accept i_d and o_d as a list with to entries.
+import numpy as np
+import pandas as pd
+import matplotlib.patches as mpatches
+from abc import ABC
+
+__all__ = [
+    "ShaftElement"
+]
+
+class Element(ABC):
+    """Element class."""
+
+    def __init__(self):
+        pass
+
+    def summary(self):
+        """A summary for the element.
+        A pandas series with the element properties as variables.
+        """
+        attributes = self.__dict__
+        attributes["type"] = self.__class__.__name__
+        return pd.Series(attributes)
+
+    # These are the abstract classes for mass, damping and stiffness matrices
+    def M(self):
+        pass
+    
+    def C(self):
+        pass
+    
+    def K(self):
+        pass
+
+class ShaftElement(Element):
+    r"""A shaft element.
+    This class will create a shaft element that may take into
+    account shear, rotary inertia an gyroscopic effects.
+    The matrices will be defined considering the following local
+    coordinate vector:
+    .. math:: [x_1, y_1, \alpha_1, \beta_1, x_2, y_2, \alpha_2, \beta_2]^T
+    Where :math:`\alpha_1` and :math:`\alpha_2` are the bending on the yz plane and
+    :math:`\beta_1` and :math:`\beta_2` are the bending on the xz plane.
+    Parameters
+    ----------
+    L : float
+        Element length.
+    i_d : float
+        Inner diameter of the element.
+    o_d : float
+        Outer diameter of the element.
+    material : ross.material
+        Shaft material.
+    n : int, optional
+        Element number (coincident with it's first node).
+        If not given, it will be set when the rotor is assembled
+        according to the element's position in the list supplied to
+        the rotor constructor.
+    axial_force : float
+        Axial force.
+    torque : float
+        Torque.
+    shear_effects : bool
+        Determine if shear effects are taken into account.
+        Default is False.
+    rotary_inertia : bool
+        Determine if rotary_inertia effects are taken into account.
+        Default is False.
+    gyroscopic : bool
+        Determine if gyroscopic effects are taken into account.
+        Default is False.

```

Comment:
Add `shear_method_calc` as an argument to the docstring.

Reply:
These changes are show in this commit [here](https://github.com/rodrigomoliveira1/ross/commit/3a53437df3e433bd71bbaf316b06eac3dc14e561)